### PR TITLE
Fix asynchronous user problems

### PR DESCRIPTION
**Pull request by YannJD**
Hello ! There were some issues with the DatabaseManager user. As most database methods are asynchronous, there was the possibility that one could start using the application before being logged in. Now authentication waits for the user to be logged in to go to another activity/fragment which should fix this issue. I also cleaned some code that wasn't necessary anymore.

### CODE REVIEW

File: app/src/main/java/com/github/ybecker/epforuml/authentication/FirebaseAuthenticator.kt [hunk 1]
```diff
@@ -33,6 +33,7 @@ class FirebaseAuthenticator(
         FirebaseAuthUIActivityResultContract()
     ) { res -> this.onSignInResult(res) }
 
+    /*
     init {
```
Comment (RomainLogean) [inline app/src/main/java/com/github/ybecker/epforuml/authentication/FirebaseAuthenticator.kt:37]: You need to decide whether to keep this code or remove this comment. 
I believe this code automatically connects you if your credentials are found. If you choose to keep it, you will likely face some issues with the profile picture (and maybe even other variables) as it is not added to the user. 
Please make sure to address this before merging!

**Review by RomainLogean — CHANGES_REQUESTED** [review]
Hello Yann,

Great job on spotting the race condition in the code! It's good thing that you fixed it, and I think the solution you found to fix it is smart.
I also appreciate that you added tests for the Account fragment. You should check the JaCoCo report to improve your coverage, but I understand that these tests can be challenging to code.
However, it seems that you have forgotten to address the code that you commented. I think this code should be in the code as it is the one you added to get connected automatically on last sprint. 

Keep up and let me know when you fixed that !

File: app/src/main/java/com/github/ybecker/epforuml/authentication/FirebaseAuthenticator.kt (hunk 1 above)
Comment (YannJD) [reply app/src/main/java/com/github/ybecker/epforuml/authentication/FirebaseAuthenticator.kt:37]: My bad ! Those lines are indeed not needed anymore as we now check and skip the login process in checkIfAlreadySignedIn in LoginActivity (there is also similar commented code in DatabaseManager that I will remove). This function is cleaner and we don't need to create a new user but just fetch it from the database.

**Review by RomainLogean — APPROVED** [review]
Everything is alright now thank you !

**Pull request merged**